spec/features/merchants/merchant_items/show_spec.rb

```diff
@@ -134,14 +134,123 @@
+    within("div#merchant-item-description") do
       expect(page).to have_content("Description: #{@item_1.description}")
+    end
+
+    within("div#merchant-item-price") do
       expect(page).to have_content("Current Price: $751.07")
     end
   end
+
+  it "There is a link to update the item that takes you to a form" do
+    visit merchant_item_path(@merchant_1, @item_1.id)
+
+    within("div#update-item") do
+      expect(page).to have_link("Update Item")
+    end
+
+    click_link("Update Item")
+
+    within("div#edit_item_form") do
+      expect(page).to have_field("Name")
+      expect(page).to have_field("Description")
+      expect(page).to have_field("Unit price")
+  end
+
+    expect(current_path).to eq(edit_merchant_item_path(@merchant_1, @item_1.id))
+  end
+
+  it "There a form that updates with the orignal info for the item" do
+    visit merchant_item_path(@merchant_1, @item_1.id)
+    click_link("Update Item")
+    expect(current_path).to eq(edit_merchant_item_path(@merchant_1, @item_1.id))
+    
+    fill_in("Name", with: "Something")
+    fill_in("Description", with: "It is very cool!")
+    fill_in("Unit price", with: "90004")
+    click_button("Update Item")
+
+    expect(current_path).to eq(merchant_item_path(@merchant_1, @item_1.id))
+
+    within("div#merchant-item-header") do
+      expect(page).to have_content("Something")
+    end
+
+    within("div#merchant-item-description") do
+      expect(page).to have_content("It is very cool!")
+    end
+
+    within("div#merchant-item-price") do
+      expect(page).to have_content("$900.04")
+    end
+  end
+
+  it "There a form that auto populates orignal info for the item" do
+    visit edit_merchant_item_path(@merchant_1, @item_1.id)
+    click_button("Update Item")
+
+    expect(current_path).to eq(merchant_item_path(@merchant_1, @item_1.id))
+
+    expect(page).to have_content("Qui Esse")
+    expect(page).to have_content("Nihil autem sit odio inventore deleniti")
+    expect(page).to have_content("$751.07")
+  end
+
+  it "There a form when You clear name box it give you an error" do
+    visit edit_merchant_item_path(@merchant_1, @item_1.id)
+
+    fill_in("Name", with: "")
+    fill_in("Description", with: "Place holder")
+    fill_in("Unit price", with: "10000")
+
+    click_button("Update Item")
+
+    expect(current_path).to eq(edit_merchant_item_path(@merchant_1, @item_1.id))
+    expect(page).to have_content("Name can't be blank")
+  end
+
+  it "There a form when You clear description box it give you an error" do
+    visit edit_merchant_item_path(@merchant_1, @item_1.id)
+
+    fill_in("Name", with: "Place Holder")
+    fill_in("Description", with: "")
+    fill_in("Unit price", with: "10000")
+
+    click_button("Update Item")
+
+    expect(current_path).to eq(edit_merchant_item_path(@merchant_1, @item_1.id))
+    expect(page).to have_content("Description can't be blank")
+  end
+
+  it "There a form when You clear Unit price box it give you an error" do
```

Comment:
231 and 244 have the same `it` block description.

Can you clarify the `it` blocks to more accurately describe what each test does?

Reply:
yes done!